libs/redisconsumer/consumer.go

```diff
@@ -114,21 +114,42 @@ func (redisClient *RedisClient) CreateStream(ctx context.Context, stream, consum
 }
 
 // AddMessage to the specified redis stream
-func (redisClient *RedisClient) AddMessages(ctx context.Context, stream string, message ...interface{}) error {
+func (redisClient *RedisClient) AddMessages(ctx context.Context, stream string, messages ...interface{}) error {
 	pipe := ((*redis.Client)(redisClient)).Pipeline()
-
-	for _, v := range message {
-		pipe.XAdd(
-			ctx, &redis.XAddArgs{
-				Stream: stream,
-				Values: map[string]interface{}{
-					"data": v}},
-		)
+	var err error
```

Comment:
This `err` here does not make sense anymore. Now errors can occur, are checked and returned from inside the loop. So you can remove this from here, and also replace the final line with `return nil`.